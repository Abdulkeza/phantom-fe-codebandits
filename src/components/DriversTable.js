import React from 'react';

import driversDB from '../database/driversDB.json';
import DriversManagement from './dropdowns/DriverManagement.js';
import ManagementTable from './ManagementTable.js';

const driversTableColumns = [
  {
    Header: 'Names',
    accessor: 'name'
  },
  {
    Header: 'Email',
    accessor: 'email'
  },
  {
    Header: 'License',
    accessor: 'license'
  },
  {
    Header: 'Assigned Bus',
    accessor: 'assigned_bus'
  },
  {
    Header: 'Assigned Route',
    accessor: 'assigned_route'
  },
  {
    Header: 'Management',
    Cell: ({ row }) => <DriversManagement id={row.original.plateNumber} />
  }
];

const DriversTable = () => {
  return (
    <ManagementTable
      tableColumns={driversTableColumns}
      database={driversDB}
      searchPlaceholder="Search drivers..."
      registerNewPath="/dashboard/driver/register"
    />
  );
};

export default DriversTable;
