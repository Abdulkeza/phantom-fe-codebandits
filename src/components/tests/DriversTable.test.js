import DriversTable from '@components/DriversTable.js';
import { render } from '@testing-library/react';
import ProviderWrapper from '@utils/TestUtil.js';
import React from 'react';

describe('Driver Table', () => {
  it('Render the table', () => {
    const elem = render(
      <ProviderWrapper children={<DriversTable />}></ProviderWrapper>
    );
    expect(elem).toMatchSnapshot();
  });
});
