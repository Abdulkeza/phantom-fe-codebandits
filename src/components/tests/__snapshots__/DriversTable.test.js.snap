// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Driver Table Render the table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex justify-between content-center pb-7 overflow-x-auto"
      >
        <div>
          <input
            class="p-1 rounded-md border-2"
            name="search"
            placeholder="Search drivers..."
            type="text"
            value=""
          />
        </div>
        
      </section>
      <table
        class="w-full overflow-x-scroll font-raleway"
        role="table"
      >
        <thead>
          <tr
            class="pb-10 "
            role="row"
          >
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Names
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Email
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              License
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Assigned Bus
            </th>
            <th
              class="text-left font-extrabold pr-5 border-b-2 pb-4"
              colspan="1"
              role="columnheader"
            >
              Management
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              Muhire Girbert
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              ctruswell1@histats.com
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              1195510237288508
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              None
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <div
                class="z-10 font-sm"
              >
                <select
                  class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                  id=""
                  name=""
                  role="select"
                >
                  <option
                    hidden=""
                    value="manage"
                  >
                    Manage
                  </option>
                  <option
                    value="vies"
                  >
                    View
                  </option>
                  
                  
                  
                  
                </select>
              </div>
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              Fabrice Hategekimana
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              driver@me.com
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              1199780030880040
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <a
                class="text-primary"
                href="/#buses"
              >
                RAB1234
              </a>
            </td>
            <td
              class="pr-5 py-4 border-b-2"
              role="cell"
            >
              <div
                class="z-10 font-sm"
              >
                <select
                  class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                  id=""
                  name=""
                  role="select"
                >
                  <option
                    hidden=""
                    value="manage"
                  >
                    Manage
                  </option>
                  <option
                    value="vies"
                  >
                    View
                  </option>
                  
                  
                  
                  
                </select>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class=" mt-3 flex justify-center w-full bg-gray-300 bg-opacity-20 "
      >
        <tfoot>
          <tr
            class="p-1 flex-row flex items-center justify-center my-2 mx-auto w-full font-raleway"
          >
            <td
              colspan="5"
            >
              <div
                class="w-full justify-center flex mx-auto flex-row items-center overflow-x-scroll"
              >
                <button
                  class="mx-2 rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                  disabled=""
                >
                  &lt;|
                </button>
                 
                <button
                  class="mx-2 font-bold font-raleway"
                  disabled=""
                >
                  Prev
                </button>
                 
                <button
                  class="mx-2 font-bold font-raleway cursor-pointer"
                  disabled=""
                >
                  Next
                </button>
                 
                <button
                  class="mx-2 rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                  disabled=""
                >
                  |&gt;
                </button>
                 
                <div
                  class="flex flex-row justify-center w-full"
                >
                  <span
                    class="inline-block mx-2"
                  >
                    Page
                     
                    <strong>
                      1
                       of 
                      1
                    </strong>
                     
                  </span>
                  <span
                    class="inline-block mx-2"
                  >
                    | Go to page:
                     
                    <input
                      class=" pl-2 outline-none border rounded-md border-primary"
                      style="width: 50px;"
                      type="number"
                      value="1"
                    />
                  </span>
                   
                  <select
                    class="px-1/2 md:px-2 font-raleway rounded-md border border-primary"
                  >
                    <option
                      value="10"
                    >
                      Show 10
                    </option>
                    <option
                      value="25"
                    >
                      Show 25
                    </option>
                    <option
                      value="50"
                    >
                      Show 50
                    </option>
                  </select>
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex justify-between content-center pb-7 overflow-x-auto"
    >
      <div>
        <input
          class="p-1 rounded-md border-2"
          name="search"
          placeholder="Search drivers..."
          type="text"
          value=""
        />
      </div>
      
    </section>
    <table
      class="w-full overflow-x-scroll font-raleway"
      role="table"
    >
      <thead>
        <tr
          class="pb-10 "
          role="row"
        >
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Names
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Email
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            License
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Assigned Bus
          </th>
          <th
            class="text-left font-extrabold pr-5 border-b-2 pb-4"
            colspan="1"
            role="columnheader"
          >
            Management
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            Muhire Girbert
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            ctruswell1@histats.com
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            1195510237288508
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            None
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <div
              class="z-10 font-sm"
            >
              <select
                class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                id=""
                name=""
                role="select"
              >
                <option
                  hidden=""
                  value="manage"
                >
                  Manage
                </option>
                <option
                  value="vies"
                >
                  View
                </option>
                
                
                
                
              </select>
            </div>
          </td>
        </tr>
        <tr
          role="row"
        >
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            Fabrice Hategekimana
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            driver@me.com
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            1199780030880040
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <a
              class="text-primary"
              href="/#buses"
            >
              RAB1234
            </a>
          </td>
          <td
            class="pr-5 py-4 border-b-2"
            role="cell"
          >
            <div
              class="z-10 font-sm"
            >
              <select
                class="py-1 px-1 font-rale font-bold bg-transparent border rounded-sm"
                id=""
                name=""
                role="select"
              >
                <option
                  hidden=""
                  value="manage"
                >
                  Manage
                </option>
                <option
                  value="vies"
                >
                  View
                </option>
                
                
                
                
              </select>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class=" mt-3 flex justify-center w-full bg-gray-300 bg-opacity-20 "
    >
      <tfoot>
        <tr
          class="p-1 flex-row flex items-center justify-center my-2 mx-auto w-full font-raleway"
        >
          <td
            colspan="5"
          >
            <div
              class="w-full justify-center flex mx-auto flex-row items-center overflow-x-scroll"
            >
              <button
                class="mx-2 rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                disabled=""
              >
                &lt;|
              </button>
               
              <button
                class="mx-2 font-bold font-raleway"
                disabled=""
              >
                Prev
              </button>
               
              <button
                class="mx-2 font-bold font-raleway cursor-pointer"
                disabled=""
              >
                Next
              </button>
               
              <button
                class="mx-2 rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                disabled=""
              >
                |&gt;
              </button>
               
              <div
                class="flex flex-row justify-center w-full"
              >
                <span
                  class="inline-block mx-2"
                >
                  Page
                   
                  <strong>
                    1
                     of 
                    1
                  </strong>
                   
                </span>
                <span
                  class="inline-block mx-2"
                >
                  | Go to page:
                   
                  <input
                    class=" pl-2 outline-none border rounded-md border-primary"
                    style="width: 50px;"
                    type="number"
                    value="1"
                  />
                </span>
                 
                <select
                  class="px-1/2 md:px-2 font-raleway rounded-md border border-primary"
                >
                  <option
                    value="10"
                  >
                    Show 10
                  </option>
                  <option
                    value="25"
                  >
                    Show 25
                  </option>
                  <option
                    value="50"
                  >
                    Show 50
                  </option>
                </select>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
